src/transformers/training_args.py

```diff
@@ -1690,6 +1720,24 @@ def __post_init__(self):
             os.environ[f"{prefix}SYNC_MODULE_STATES"] = self.fsdp_config.get("sync_module_states", "true")
             os.environ[f"{prefix}USE_ORIG_PARAMS"] = self.fsdp_config.get("use_orig_params", "true")
 
+        if isinstance(self.accelerator_config, str):
+            self.accelerator_config = AcceleratorConfig.from_json_file(self.accelerator_config)
+        if self.dispatch_batches is not None:
```

Comment:
I think we need to handle the case when `accelerator_config` is `None`, in particular when the deprecated arguments are passed in

Reply:
I think just having an `elif` should be fine since we override `self.accelerator_config` if they are passed